### ISSUE TITLE
false negative when calling two polymorphic functions with omitted type arguments

### ISSUE THREAD

**Issue by theosotr** [typetools/checker-framework#6832]
### Command

```
javac -J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
  -J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED \
-processor org.checkerframework.checker.nullness.NullnessChecker \
-cp checker-framework/checker/dist/checker.jar Test.java
```

### File

```java
import java.util.*;
import java.util.function.*;
import org.checkerframework.checker.nullness.qual.*;


class A {
    public <T> @Nullable T m(T x) {
        return null;
    }

}

public class Test {
    public static <T> void invokeAny(Collection<Supplier<T>> p) {
        p.toString();

    }

    public static void main(String[] args) {
        A x = new A();
        invokeAny(x.m(new LinkedList<Supplier<String>>()));

    }
}
```

### Actual behavior

The code passes the checks, but there's NPE at runtime

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.util.Collection.toString()" because "<parameter1>" is null
	at Test.invokeAny(Test.java:15)
	at Test.main(Test.java:21)
```

### Expected behavior

The code should have been rejected.
